File: 5th_sem_C_Java_VB/Java/Program_with_numbers/binarysearch.java
import java.io.*;
class binarysearch
{
  public static void main(String args[])throws IOException
  {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    int a[] = new int[20];
    int id[] = new int[20];
    int n,i,num,l,u,m,flag=0,d=0;
    System.out.print("\nEnter number of elements : ");
    n=Integer.parseInt(br.readLine());
    l=0;
    u=n-1;
    for(i=0;i<n;i++)
    {
      System.out.print("\nEnter number "+(i+1)+ " : ");
      a[i]=Integer.parseInt(br.readLine());
      id[i]=i+1;
    }
    for(i=0;i<n;i++)
      System.out.print(a[i]+"\t");
    System.out.print("\nEnter number to be searched : ");
    num=Integer.parseInt(br.readLine());
    while(l<=u)
    {
      m=(l+u)/2;
      if(a[m]==num)
      {
        flag=1;
        d=id[m];
      }
      if(num>a[m])
        l=m+1;
      else
        u=m-1;
    }
    if(flag==1)
      System.out.println("\nNumber "+num+" found at position "+d);
    else
      System.out.println("\nNumber not found");
  }
}    

File: 6th_sem/MATLAB/From Ankita/6-2-18/binary_search.m
function binary_search(a,id,n,num)
count=0;
hi=n;
low=1;
mid=fix((hi+low)/2);
    while(hi>low && count==0)
       if num<a(mid) 
           hi=mid-1;
       else
           if num>a(mid)
               low=mid+1;
           else
               fprintf('\n%d found at position %d',num,id(mid));
               count=count+1;
               i=mid-1;
               while(i>=1)
                   if a(i)==num
                       fprintf('\n%d found at position %d',num,id(i));
                       count=count+1;
                       i=i-1;
                       if i==0
                           break
                       end
                   else
                       break
                   end
               end
               i=mid+1;
               while(i<=n)
                   if a(i)==num
                       fprintf('\n%d found at position %d',num,id(i));
                       count=count+1;
                       i=i+1;
                       if i==n
                           break
                       end
                   else
                       break
                   end
               end
           end
       end
    end
    if count~=0
        fprintf('\nNumber of times %d found in your list=%d\n',num,count);
    else
        fprintf('\n%d not found in the list.',num);
    end
end
